mlir/lib/Dialect/QCO/IR/QCOUtils.cpp

```
@@ -22,53 +23,133 @@
+  const auto q1 = lookupWireId(wireIds, unitary.getInputQubit(1));
+  if (!q0.has_value() || !q1.has_value()) {
+    return std::nullopt;
+  }
+  if (numTargets == 2 && *q0 == 0 && *q1 == 1) {
+    return unitary.getUnitaryMatrix<DynamicMatrix>();
+  }
+  return unitary.getUnitaryMatrix<Matrix4x4>()->embedInNqubit(numTargets, *q0,
+                                                              *q1);
+}
+
+std::optional<DynamicMatrix> composeNTargetBodyMatrix(Block& block,
```

Cppcheck (2.21.0)

[style] The function 'composeNTargetBodyMatrix' is never used.

(unusedFunction)